discovery/consul/consul_test.go

```diff
@@ -397,3 +399,73 @@ func TestGetDatacenterShouldReturnError(t *testing.T) {
 		require.Equal(t, "", d.clientDatacenter)
 	}
 }
+
+func TestUnmarshalConfig(t *testing.T) {
+	unmarshal := func(d []byte) func(interface{}) error {
+		return func(o interface{}) error {
+			return yaml.Unmarshal(d, o)
+		}
+	}
+
+	cases := []struct {
+		name       string
+		config     string
+		expected   SDConfig
+		errMessage string
+	}{
+		{
+			name: "good",
+			config: `
+server: localhost:8500
+username: 123
+password: 1234
+`,
+			expected: SDConfig{
+				Server: "hi",
+				HTTPClientConfig: config.HTTPClientConfig{
+					BasicAuth: &config.BasicAuth{
+						Username: "hello",
```

Comment:
That should be 123 ?

Reply:
Yeah, I forgot to check the expected config.